### ISSUE TITLE
Split up `astconv.rs`

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/rust#67418]
That file (https://github.com/rust-lang/rust/blob/dab390683196e31116cb61ccf3eec1a4af729def/src/librustc_typeck/astconv.rs) has almost 3k LOC and just 3 lines of documentation on the file (methods and types are documented though). I'm entirely unsure how to process that file. I think splitting it up into submodules (e.g. start by extracting out all functions just handling error reporting into a separate module).

Some of this work can be just mechanical splitting up, but some parts may require you to start understanding at least some of what's going on in the file in order to find a good module structure.

cc @Centril

**Comment by TheSamsa** [comment]
I'll do it, although I am not familiar with the `typeck`.
At least the mechanical part is no problem. As to the model structure, I am going to dive into that issue so I can better think of a reasonable structure.

If it is a problem that I am not familiar with the topic, I will back down, no worries.

**Comment by oli-obk** [comment]
That's awesome. Please wait until https://github.com/rust-lang/rust/pull/67268 is merged though (you can subscribe to it so you see when it does get merged), because otherwise you'll just get loads of conflicts.

**Comment by oli-obk** [comment]
If you are unsure about the module structure, you can first post a design suggestion here if you want.

**Comment by spadaval** [comment]
@TheSamsa You still working on this?

**Comment by TheSamsa** [comment]
Not quite, my other PR is not yet through and I thought I wait for it.

But if you want to do it, just take it.
If not I'll be on it soon.

**Comment by Areredify** [comment]
I would like to claim this, if nobody minds

**Comment by oli-obk** [comment]
@rustbot assign @Areredify 

It's all yours!

**Comment by CohenArthur** [comment]
I'd like to work in this if @Areredify is not on it anymore @Dylan-DPC

**Comment by Dylan-DPC** [comment]
@CohenArthur generally it's fine if it's 3 months and there is no pr from the author. you can start working on it

**Comment by CohenArthur** [comment]
Gotcha, thanks
